### PR TITLE
Add support for virtualenv steps

### PR DESCRIPTION
**Pull request by joshuaspence**
Adds support for the https://wiki.jenkins-ci.org/display/JENKINS/ShiningPanda+Plugin plugin.

### CODE REVIEW

**Comment by daspilker** [comment]
Add documentation and tests for the all options.

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/step/StepContext.groovy [hunk 1]
```diff
@@ -723,6 +723,42 @@ class StepContext extends AbstractExtensibleContext {
...
 
     /**
+     * @since 1.36
+     */
+    @RequiresPlugin(id = 'shiningpanda')
+    void python(@DslContext(PythonContext) Closure closure) {
+        PythonContext context = new PythonContext()
+        ContextHelper.executeInContext(closure, context)
+
+        stepNodes << new NodeBuilder().'jenkins.plugins.shiningpanda.builders.PythonBuilder' (plugin: 'shiningpanda') {
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/step/StepContext.groovy]: remove the plugin attributes

Comment (joshuaspence) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/step/StepContext.groovy]: What do you mean exactly?

Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/step/StepContext.groovy]: The plugin attribute `(plugin: 'shiningpanda')`, the DSL does not generate the plugin attributes for any plugin.

File: docs/Job-reference.md [hunk 2]
```diff
@@ -4183,6 +4185,33 @@ See [Exclusion Resources](#exclusion-resources).
...
 (since 1.24)
 
+### Python
+
+```groovy
+job {
+    steps {
+        python {
+            command(String command)
+            ignoreExitCode(boolean ignoreExitCode)
```
Comment (daspilker) [inline docs/Job-reference.md]: Make the boolean arguments optional, with a default of `true`.

File: docs/Job-reference.md [hunk 3]
```diff
@@ -4183,6 +4185,33 @@ See [Exclusion Resources](#exclusion-resources).
...
+            nature(String nature)
+            pythonName(String pythonName)
+        }
+
+        virtualenv {
+            clear(boolean clear)
+            command(String command)
+            ignoreExitCode(boolean ignoreExitCode)
+            name(String name)
+            nature(String nature)
```
Comment (daspilker) [inline docs/Job-reference.md]: Mention the possible values for `nature`.

File: docs/Job-reference.md [hunk 4]
```diff
@@ -4183,6 +4185,33 @@ See [Exclusion Resources](#exclusion-resources).
...
+            pythonName(String pythonName)
+            systemSitePackages(boolean systemSitePackages)
+        }
+    }
+}
+```
+
```
Comment (daspilker) [inline docs/Job-reference.md]: Add an example.

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/step/StepContext.groovy [hunk 5]
```diff
@@ -748,6 +748,42 @@ class StepContext extends AbstractExtensibleContext {
     }
 
     /**
+     * @since 1.38
+     */
+    @RequiresPlugin(id = 'shiningpanda')
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/step/StepContext.groovy]: Add a minimum version, also for `virtualenv` below.

**Comment by daspilker** [comment]
OK, great. Fix the last issues and it's good to merge.

**Pull request merged**